app_test_suite/cluster_providers/kind_cluster_provider.py

```diff
@@ -83,3 +100,16 @@ def delete_cluster(self, cluster_info: cluster_provider.ClusterInfo) -> None:
             raise ATSTestError(f"Error when deleting KinD cluster. Exit code is: {run_res.returncode}")
         os.remove(kube_config_path)
         logger.info("KinD cluster deleted successfully")
+
+    def augment_kind_config_file(self, kind_config_path: str, image_override: str) -> str:
+        with open(kind_config_path, "r") as file:
+            config_yaml = yaml.safe_load(file)
+            config_yaml["nodes"] = [dict(node, **{"image": image_override}) for node in config_yaml["nodes"]]
```

Comment:
This piece seems pretty hard to read - at least to me. WDYT about replacing with this:

```python
for node in config_yaml["nodes"]:
    node.update({"image": image_override})
```

Reply:
Yeah much better!

Reply:
```suggestion
            config_yaml["nodes"] = [dict(node, **{"image": image_override}) for node in config_yaml["nodes"]]
            for node in config_yaml["nodes"]:
                node.update({"image": image_override})
```